### PR TITLE
Feature/trcl 2763 optimization

### PR DESCRIPTION
**Pull request by johnqh**
Use Double for everything except for raw data from the protocol.
Parser.asDouble() will handle super long string which would normally overflow double. (Have happened with indexPrice in v3)

### CODE REVIEW

File: src/commonTest/kotlin/exchange.dydx.abacus/utils/Map+Satisfy.kt [hunk 1]
```diff
@@ -84,28 +85,38 @@ private fun IMap<String, Any>.satisfiesLists(
...
     path: String
 ) {
-    parser.asDouble(value1)?.let {
-        assertEquals(parser.asDouble(value2), it, "$path should have a value of $it")
-        return
+    parser.asDouble(value2)?.let {
+        val value = parser.asDouble(value1)
+        if (value != null) {
+            val stepSize = parser.asDouble(it.tickDecimals())!!
+            val roundedValue = Rounder.round(parser.asDouble(value2)!!, stepSize)
```
Comment (ruixhuang) [inline src/commonTest/kotlin/exchange.dydx.abacus/utils/Map+Satisfy.kt:96]: We check null for value1.  Should we check null for value2 instead of "!!" ?

**Comment by rosepuppy** [comment]
Looks like the version needs to be bumped again

File: src/commonMain/kotlin/exchange.dydx.abacus/utils/Parser.kt [hunk 2]
```diff
@@ -86,11 +86,16 @@ class Parser : ParserProtocol {
 
         val string = data as? String
         if (string != null) {
+            val withDecimalPoint = string.replace(",", ".")
```
Comment (rosepuppy) [inline src/commonMain/kotlin/exchange.dydx.abacus/utils/Parser.kt:89]: What is the reason for this this?

**Review by rosepuppy — APPROVED** [review]
Rui should approve as well, but LGTM

**Review by ruixhuang — APPROVED** [review]

**Pull request merged**